src/flow/ui/screens/LoadingScreen.tsx

```diff
@@ -5,11 +5,7 @@ import ScreenView from '../../../components/ScreenView';
 
 const LoadingScreen = () => {
   return (
```

Comment:
[2/3]

The naive thought is that only Loading screen needs to be tweaked so that on mount, it has to read the context value.

```ts
const LoadingScreen = () => {
  const { initialRouteName } = useInitialRouteNameStateContext(InitialRouteNameStateContext);
  const navigation = props.navigation;

  useEffect(() => {
    // Jump into the stack which has the entry point we want, not the route inside that stack.
    navigation.navigate(initialRouteName ? initialRouteName + 'Entry' : LoadingScreen.ROUTE);
  }, [initialRouteName]);

  return (
```

The realization is that there are a few critical problems which I'm unfortunately struggling to wrap my head around. Roughly can be explained below (poorly written as I didn't go to deep here):
1. The Loading screen scope is too constrained to control the whole flow by itself. When the stack leaves this screen, it unmounts, and the only way to return to it  is to tell React Navigation to go back to it. This approach, so far, was attempted only at the Logout screen, which means that there are blind spots, particularly in the Android native side, which also manipulates the initial route but has no awareness of how to deal with this.
2. When going back to the Loading screen, the component immediately runs the `useEffect` block, meaning that as soon as it mounts, it runs the automatic navigation, when we expected that it would wait for the Landing screen (which would be currently in view) to disappear first before running navigation.
3. Sticking to this approach is likely to mean that the Loading screen needs to be aware of whether the Landing screen is visible or invisible, which is yet another state. The potential risk I can think of is to run into race conditions. Bridging is not synchronous, meaning that there's no guarantee that having this state will allow the app to navigate consistently.